### DISCUSSION TITLE
How to change HomeKit pairing code ?

### DISCUSSION THREAD

**Discussion by Blueberry94** [RavenSystem/esp-homekit-devices#1876] [Q&A] [answered]
Hey, 

I successfully flashed the haa firmware and was able to connect to HomeKit. My device is working :)

But for setting up a second device I want to change the HomeKit pairing code which is set by default to: 021-82-017
How to change it for example to 123-45-678 ? 

Thanks a lot :) 

Best

**Comment by Daveblanche** [comment]
You don’t need to change the pairing code as the pairing ID is set during the initial on boarding with HomeKit. I’ve got 22 (i think) HAA devices on my network all using the same code. No need to change. 
You do, however, need to reset the HomeKit ID in the HAA setup mode if you update the JSON code or remove it from your home and want to re-add, generally anyway.

**Reply by Blueberry94** [reply]
Thanks for the fast answer :) But does it work to change the pairing code in any way? Where is it exactly set in the code? Do I need to compile modified firmware? 

Could you explain it?

Thanks a lot :)

**Comment by Daveblanche** [comment]
You lost me after pairing code.
I use this stuff for lots of things all around my house. It’s brilliant, and the community is superb.
But I’ve never compiled, only used the firmware offered as .bin.

**Comment by RavenSystem** [comment, marked as answer]
Changing pairing code os illegal: 
https://github.com/RavenSystem/esp-homekit-devices/blob/master/LICENSE

**Reply by Blueberry94** [reply]
Thanks :) Did not saw it!

Best
